lib/computeStatistics.js

```diff
@@ -0,0 +1,182 @@
+            case 3: // LINE_STRIP
+                numberOfRenderedPrimitives += Math.max(indicesCount - 1, 0);
+                break;
+            case 4: // TRIANGLES
+                numberOfRenderedPrimitives += (indicesCount / 3);
+                break;
+            case 5: // TRIANGLE_STRIP
+            case 6: // TRIANGLE_FAN
+                numberOfRenderedPrimitives += Math.max(indicesCount - 2, 0);
+                break;
+        }
+    }
+
+    return {
+        numberOfDrawCalls : numberOfDrawCalls,
+        numberOfRenderedPrimitives : numberOfRenderedPrimitives
+    };
+}
+
+/**
+ * Returns an object containing the statistics for the glTF assest.
+ *
+ * @param {Object} gltf A javascript object containing a glTF asset with extras and defaults.
+ * @returns {Statistics} Object containing the properties of the assest.
+ *
+ * @see Statistics
+ * @see printStatistics
+ */
+function getStatistics(gltf) {
+    if (!defined(gltf)) {
+        return new Error('gltf must be defined');
+    }
+
+    var drawStats = getDrawCallStatistics(gltf);
+
+    var statistics = new Statistics();
+    statistics.buffersSizeInBytes          = getBuffersSize(gltf.buffers);
+    statistics.numberOfImages              = getNumberOfProperties(gltf.images);
+    statistics.numberOfExternalRequests    = getNumberOfExternalRequests(gltf);
+    statistics.numberOfDrawCalls           = drawStats.numberOfDrawCalls;
+    statistics.numberOfRenderedPrimitives  = drawStats.numberOfRenderedPrimitives;
+    statistics.numberOfNodes               = getNumberOfProperties(gltf.nodes);
+    statistics.numberOfMeshes              = getNumberOfProperties(gltf.meshes);
+    statistics.numberOfMaterials           = getNumberOfProperties(gltf.materials);
+    statistics.numberOfAnimations          = getNumberOfProperties(gltf.animations);
+
+    return statistics;
+}
+
+/**
+ * Print the glTF statistics onto the console.
+ *
+ * @param {Object} gltf A javascript object containing a glTF asset with extras and defaults.
+ * @param {String} gltfPath The path of the glTF file being analysed.
+ *
+ * @see Statistics
+ * @see getStatistics
+ */
+function printStatistics(gltf, gltfPath) {
+    var statistics = getStatistics(gltf);
+    var output =
+        '\nKey statistics for ' + gltfPath +
+        '\n--------------\n' +
+        '\nTotal size of all buffers: ' + statistics.buffersSizeInBytes + ' bytes' +
+        '\nImages: ' + statistics.numberOfImages +
+        '\nExternal requests (not data uris): ' + statistics.numberOfExternalRequests +
+        '\n' +
+        '\nDraw calls: ' + statistics.numberOfDrawCalls +
+        '\nRendered primitives (e.g., triangles): ' + statistics.numberOfRenderedPrimitives +
+        '\n' +
+        '\nNodes: ' + statistics.numberOfNodes +
+        '\nMeshes: ' + statistics.numberOfMeshes +
+        '\nMaterials: ' + statistics.numberOfMaterials +
+        '\nAnimations: ' + statistics.numberOfAnimations;
+
+    console.log(output);
+}
```

Comment:
I'm starting to think... now that the `Statistics` class exists, this could be a function of the class and `computeStatistics` could be renamed `getStatistics` and just expose the `getStatistics` function. `printStatistics` could instead become `toString` and the caller would be responsible in calling `console.log`. Does this seem like a good idea?